src/interface.jl

```diff
@@ -135,15 +136,16 @@ function jacobian!(obj, x)
 end
 function jacobian!!(obj, x)
     obj.df(obj.DF, x)
-    copy!(obj.x_df, x)
-    obj.df_calls .+= 1
+    copyto!(obj.x_df, x)
+    (obj.df_calls .+= 1; nothing)
```

Comment:
Ah, sorry, you're write. I was adding the `nothing` so that Julia doesn't use the return value:
```julia
julia> function g!(a,b,c)
           a .*= b .+ c
       end
┌ Warning: Deprecated syntax `using the value of `.=`` around REPL[20]:2.
└ @ nothing REPL[20]:2
g! (generic function with 1 method)
julia> function g2!(a,b,c)
           a .*= b .+ c; nothing
       end
g2! (generic function with 1 method)
```
but then I didn't want to change the actual behavior of the function, so I went back and repeated the last line -- but then Julia knows the return isn't being used, and doesn't emit the warnings. That is, this is fine and preserves the behavior of the old function:
```julia
 function value_jacobian!!(obj, F, J, x)
    obj.fdf(F, J, x)
    copyto!(obj.x_f, x)
    copyto!(obj.x_df, x)
     obj.f_calls .+= 1
    obj.df_calls .+= 1
    obj.df_calls
 end
```
Also, the semicolon `; nothing` is unnecessary. Cleaner to just use a new line. 
I'll go back and fix all these.

Not sure if the return value is ever used, but I wanted to err on the side of preserving behavior.